bool UpArrow
bool DownArrow
bool LeftArrow
bool RightArrow
bool x
